### PR TITLE
Use do-not-merge/needs-foo labels for kubernetes/kubernetes

### PR DESCRIPTION
**Pull request by spiffxp**
Currently there are two innocuously grey labels `needs-kind` and
`needs-sig` which can block merges to kubernetes/kubernetes.  They do
not use the same `do-not-merge/` prefix used by other merge-blocking
labels, nor are they colored red.

This is the first step in migrating to `do-not-merge/needs-foo` variants
of these labels:

- add the new labels for kubernetes/kubernetes
- update tide and fejta-bot queries to exclude the new labels
- update require-matching-label plugin config to apply the new labels
  going forward

Because the grey versions of these labels are used by other repos for
issue triage, the old labels don't need to be removed via automation.
They will be manually removed from kubernetes/kubernetes PRs.

This should be most of https://github.com/kubernetes/test-infra/issues/19805

### CODE REVIEW

**Comment by spiffxp** [comment]
/test pull-test-infra-bazel
image pull flake ```Pull command failed: 2020/11/03 21:03:23 Running the Image Puller to pull images from a Docker Registry...
2020/11/03 21:03:24 Image pull was unsuccessful: unable to save remote image gcr.io/cloud-builders/gcloud@sha256:5b49dfb5e366dd75a5fc6d5d447be584f8f229c5a790ee0c3b0bd0cf70ec41dd: unable to write image layers: unable to write image layer: unable to write the contents of layer 0 to /bazel-scratch/.cache/bazel/_bazel_root/05618a594cb3499a4b912817865867cd/external/gcloud-base/image/000.tar.gz: read tcp 10.60.119.139:39678->108.177.112.128:443: read: connection reset by peer
```

**Comment by spiffxp** [comment]
FYI @kubernetes/release-team-leads
FYI @kubernetes/sig-release-leads 
/cc @cblecker @nikhita 
contribex TL's

**Comment by cblecker** [comment]
/lgtm
/hold

Looks fine from contribex perspective. Makes things clearer.
cc @nikhita 

Would like thumbs up from release folks too

**Review by fejta — APPROVED** [review]

**Comment by justaugustus** [comment]
From SIG Release:
/lgtm
/approve

FYI @kubernetes/sig-release-leads @kubernetes/release-team

**Review by hasheddan — APPROVED** [review]
/lgtm

**Comment by spiffxp** [comment]
End of my day, I'll remove hold tomorrow when I can monitor

**Comment by mrbobbytables** [comment]
/lgtm
I definitely think it'll help convey they are blocking better.

**Comment by dims** [comment]
sounds like a good plan 👍  LGTM

**Review by nikhita — COMMENTED** [review]
One nit, but lgtm otherwise. :+1: 

Just for the record - once we manually remove the `needs-sig` and `needs-kind` labels from PRs in k/k, the `required-matching-label` plugin will automatically add the `do-not-merge/needs-sig` and `do-not-merge/needs-kind` labels since it [reacts on label changes](https://github.com/kubernetes/test-infra/blob/e3383effbefcb527a499df74c53f3d2b4341206b/prow/plugins/require-matching-label/require-matching-label.go#L42).

File: label_sync/labels.yaml [hunk 1]
```diff
@@ -1019,6 +1019,19 @@ repos:
...
+      - color: e11d21
+        description: Indicates a PR lacks a `kind/foo` label and requires one.
+        name: do-not-merge/needs-kind
+        target: prs
+        prowPlugin: require-matching-label
+        addedBy: prow
+      - color: e11d21
+        description: Indicates an issue or PR lacks a `sig/foo` label and requires one.
+        name: do-not-merge/needs-sig
+        target: both
```
Comment (nikhita) [inline label_sync/labels.yaml:1031]: Oops looks like I didn't really click on the `Start Review` option in the UI :woman_facepalming: 

nit: do-not-merge doesn't have any meaning for issues :) 

```suggestion
        target: prs
```

**Review by hasheddan — APPROVED** [review]
/lgtm

**Pull request merged**